new-app/src/App/runGame/OTPInputGuess.tsx

```diff
@@ -52,9 +59,35 @@ const OTPInputGuess = ({guess, setGuess}: {guess: any[], setGuess: React.Dispatc
     submitGuess();
   };
 
+  const buildDigitStyle = (hint?: string): string => {
```

Comment:
Maybe instead of having the hint be optional, type the hint as
`type Hint= "miss" | "key" | "neither"`
and then you can switch on these without getting a no return error.